sklearn/ensemble/_gb.py

```diff
@@ -554,7 +564,10 @@ def fit(self, X, y, sample_weight=None, monitor=None):
             self.train_score_ = self.train_score_[:n_stages]
             if hasattr(self, "oob_improvement_"):
                 self.oob_improvement_ = self.oob_improvement_[:n_stages]
-
+            if hasattr(self, "oob_scores_"):
+                self.oob_scores_ = self.oob_scores_[:n_stages]
+            if hasattr(self, "oob_score_"):
+                self.oob_score_ = self.oob_scores_[-1]
```

Comment:
We don't actually need to check for the attribute here because if `oob_improvement_` exist then the scores.

```suggestion
                self.oob_scores_ = self.oob_scores_[:n_stages]
                self.oob_score_ = self.oob_scores_[-1]
```

It could be worth to add a small comment under the `if` line: `# OOB scores were computed`